### ISSUE TITLE
Convert between Symbol and AbstractString

### ISSUE THREAD

**Issue by oxinabox** [JuliaLang/julia#35102]
It would be handy if `convert(::Type{Symbol}, str::AbstractString)` and `convert(::Type{String}, sym::Symbol)` worked
  
related #35033 

`Symbol` isn't a string, but it is string-like.
and we can loselessly convert between `Symbol` and `String`

**Comment by KristofferC** [comment]
Isn't it undesirable to implicitly convert things to each other when they are non-equal. For example:

```
julia> struct Foo a::String end

julia> Base.convert(::Type{String}, sym::Symbol) = string(sym)

julia> :a == "a"
false

julia> Foo(:a) == Foo("a")
true
```

**Comment by oxinabox** [comment]
Unless they are numbers?
```julia
julia> struct Foo a::Float16 end

julia> 0.000_000_10 == 0.000_000_11
false

julia> Foo(0.000_000_10) == Foo(0.000_000_11)
true
```

**Comment by KristofferC** [comment]
Yeah, lossy conversion for floats feels like a special case (as we know floats have quite a lot of weirdness). And you at least have the case that they are in some kind of the same domain since they can be equal.

But considering

> Symbol isn't a string, but it is string-like and we can loselessly convert between Symbol and String

perhaps you also want to make `:a == "a"`?

**Comment by oscardssmith** [comment]
That sounds really dangerous. It might be nice if `:a ≃ "a"` This might be worse though.

**Comment by oxinabox** [comment]
> Yeah, lossy conversion for floats feels like a special case 

That is fair.

>   perhaps you also want to make :a == "a"?

Maybe we do. Feels pretty wild, but maybe we do.
We would not to `isequal` though that would be unreasonable right?

Such a rabbit hole.
if going this wild.
maybe we want `Symbol <: AbstractString`

**Comment by StefanKarpinski** [comment]
Symbol's aren't strings. They are the [representation of identifiers as data](https://stackoverflow.com/questions/23480722/what-is-a-symbol-in-julia). In particular, if you `eval` a string, you get the same string back, whereas if you eval a symbol, you get the value that symbol is bound to in the evaluation environment. Having two things that are equal behave so radically differently under such an operation would not be a good idea.

**Comment by oxinabox** [comment]
Yeah, this feels too wild.

In particular I think the fact that symbols are conseptually indivisible they are "atoms", makes treating them as strings wrong.
Where as strings are free monads, symbols are not.
